frontend/packages/yki/src/components/reassessment/evaluationOrder/PublicEvaluationOrderForm.tsx

```diff
@@ -0,0 +1,293 @@
+        <b>{ExamUtils.languageAndLevelText(evaluationPeriod)}</b>
+      </Text>
+      <Text>
+        {translateCommon('examDate')}:{' '}
+        <b>{DateUtils.formatOptionalDate(evaluationPeriod.exam_date)}</b>
+      </Text>
+    </div>
+  );
+};
+
+const ExaminationPartCheckbox = ({
+  examinationPart,
+}: {
+  examinationPart: keyof ExaminationParts;
+}) => {
+  const translateCommon = useCommonTranslation();
+  const { examinationParts } = useAppSelector(evaluationOrderSelector);
+  const dispatch = useAppDispatch();
+
+  return (
+    <FormControlLabel
+      control={
+        <Checkbox
+          checked={examinationParts[examinationPart]}
+          onChange={(_, checked) =>
+            dispatch(setExaminationParts({ [examinationPart]: checked }))
+          }
+        />
+      }
+      label={`${translateCommon('examParts.' + examinationPart)} 50 €`}
+    />
+  );
+};
+
+const SelectExaminationParts = () => {
+  const { t } = usePublicTranslation({
+    keyPrefix: 'yki.component.evaluationOrderForm.selectExaminationParts',
+  });
+  const { examinationParts } = useAppSelector(evaluationOrderSelector);
+  const sumTotal = Object.values(examinationParts).reduce(
+    (acc, val) => (val ? acc + 50 : acc),
+    0
+  );
+
+  return (
+    <>
+      <H2>{t('heading')}</H2>
+      <FormControl component="fieldset" variant={TextFieldVariant.Standard}>
+        <Typography variant="h3" component="legend">
+          {t('selectParts')}
+        </Typography>
+        <Text>{t('selectAtLeastOne')}</Text>
+        <FormGroup>
+          <ExaminationPartCheckbox examinationPart="readingComprehension" />
+          <ExaminationPartCheckbox examinationPart="speechComprehension" />
+          <ExaminationPartCheckbox examinationPart="writing" />
+          <ExaminationPartCheckbox examinationPart="speaking" />
+        </FormGroup>
+      </FormControl>
+      <Text>
+        {t('sumTotal')}: <b>{sumTotal} €</b>
+      </Text>
+    </>
+  );
+};
+
+const PayerDetailsTextField = ({
+  field,
+}: {
+  field: keyof Omit<PayerDetails, 'birthdate'>;
+}) => {
+  const { t } = usePublicTranslation({
+    keyPrefix:
+      'yki.component.evaluationOrderForm.fillPayerDetails.placeholders',
+  });
+  const translateCommon = useCommonTranslation();
+  const value = useAppSelector(evaluationOrderSelector).payerDetails[field];
+  const showErrors = useAppSelector(evaluationOrderSelector).showErrors;
+  const dispatch = useAppDispatch();
+
+  const getFieldError = useCallback(
+    (field: keyof Omit<PayerDetails, 'birthdate'>) => {
+      if (showErrors) {
+        const fieldType =
+          field === 'email' ? TextFieldTypes.Email : TextFieldTypes.Text;
+        const error = InputFieldUtils.inspectCustomTextFieldErrors(
+          fieldType,
+          value,
+          true
+        );
+
+        if (error) {
+          return translateCommon(error);
+        } else {
+          return null;
+        }
+      } else {
+        return null;
+      }
```

Comment:
```suggestion
      if (!showErrors) {
        return null;
      }
      
      const fieldType =
        field === 'email' ? TextFieldTypes.Email : TextFieldTypes.Text;
      const error = InputFieldUtils.inspectCustomTextFieldErrors(
        fieldType,
        value,
        true
      );

      if (!error) {
        return null;
      }
      
      return translateCommon(error);
```

Tällä vois vähän yksinkertaistaa näitä ehtoja.